### PR TITLE
Find cell bug

### PR DESCRIPTION
**Pull request by joakim-hove**
- The cell_contains function will now use signed distance from planes
  defined by 12 surface triangles on the cell faces.

- The implementation will check for cell twist, and will return false
  for all cells with nonzero twist.

- The testing of cell_contains uses function which guarantees test point
  is inside cell, and not center point.

### CODE REVIEW

**Comment by pgdr** [comment]
@joakim-hove sorry for not picking up on this.  I'm reviewing this now.

File: libecl/src/ecl_grid.c [hunk 1]
```diff
@@ -1066,6 +1066,21 @@ static void ecl_cell_taint_cell( ecl_cell_type * cell ) {
...
+  int twist_count = 0;
+
+  for (int c = 0; c < 4; c++) {
+    const point_type * p1 = &cell->corner_list[c];
+    const point_type * p2 = &cell->corner_list[c + 4];
+    if (sign * (p2->z - p1->z) < 0)
+      twist_count += 1;
+  }
+  return twist_count;
+}
```
Comment (pgdr) [inline libecl/src/ecl_grid.c]: Aren't you going to use `sign`?  As it is now, the `if` statement is essentially `if (p1->z  > p2->z) twist_count++`.

Comment (pgdr) [inline libecl/src/ecl_grid.c]: yes, I understand that in principle, `sign` could be `-1` however, that's not going to happen unless we explicitly add some code somewhere somewhen.   Why not stick with the simpler version until we implement support for changing sign?

File: libecl/src/ecl_grid.c [hunk 2]
```diff
@@ -3865,43 +3894,60 @@ bool ecl_grid_cell_contains_xyz1( const ecl_grid_type * ecl_grid , int global_in
...
             plane_nr++;
-            if (plane_nr == 6)
-              return true;
           }
         }
-      } else
-        return false;
+      }
+
+      return false;
```
Comment (pgdr) [inline libecl/src/ecl_grid.c]: This `while`-loop could preferably be written as a `for` loop.  Something like this?
```c
for (int plane_nr = 0; plane_nr < 12; plane_nr++) {
  p0 = &cell->corner_list[ bounding_planes[plane_nr][0] ];
  p1 = &cell->corner_list[ bounding_planes[plane_nr][1] ];
  p2 = &cell->corner_list[ bounding_planes[plane_nr][2] ];

  if (point_equal(p0, p1) || point_equal(p0,p2) || point_equal(p1,p2))
    continue;

  if (sign * point3_plane_distance(p0 , p1 , p2 , &p ) < 0)
    return false;
}
return true;
```

File: libecl/src/ecl_grid.c [hunk 3]
```diff
@@ -4472,6 +4522,27 @@ void ecl_grid_get_xyz3(const ecl_grid_type * grid , int i, int j , int k, double
...
+  return true;
+}
+*/
+
+/*bool ecl_grid_get_xyz_inside3(const ecl_grid_type * grid , int i, int j , int k, double *xpos , double *ypos , double *zpos) {
+  const int global_index = ecl_grid_get_global_index__(grid , i , j , k );
+  return ecl_grid_get_xyz_inside1( grid , global_index , xpos, ypos, zpos );
+}
+*/
+
```
Comment (pgdr) [inline libecl/src/ecl_grid.c]: delete this code?  or keep it if you strongly feel like it.

File: libecl/tests/ecl_grid_cell_contains.c [hunk 4]
```diff
@@ -183,12 +235,12 @@ void test_find( ecl_grid_type * grid ) {
...
+//
 //              if (find_index >= 0)
 //                find_cell_inside_out = ecl_grid_cell_inside_out1( grid , find_index );
 //              else
 //                test_assert_int_not_equal( -1 , find_index );
-//              
+//
 //              if (find_cell_inside_out == init_cell_inside_out) {
 //                if (init_cell_inside_out == false)
 //                  test_assert_int_equal( init_index , find_index );
```
Comment (pgdr) [inline libecl/tests/ecl_grid_cell_contains.c]: delete all of this code?

File: libecl/tests/ecl_grid_cell_contains_wellpath.c [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/*
+   Copyright (C) 2014  Statoil ASA, Norway.
+
+   The file 'ecl_grid_cell_contains.c' is part of ERT - Ensemble based
```
Comment (pgdr) [inline libecl/tests/ecl_grid_cell_contains_wellpath.c]: why not always use "This file is part of ERT"?

File: libecl/tests/ecl_grid_cell_contains_wellpath.c [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  return expected;
+}
+
+
+
+int main(int argc , char ** argv) {
+  util_install_signals();
+  {
+    ecl_grid_type * grid = ecl_grid_alloc( argv[1] );
+    vector_type * expected = load_expected( argv[2] );
```
Comment (pgdr) [inline libecl/tests/ecl_grid_cell_contains_wellpath.c]: add `test_assert_int_equal(10 , vector_get_size(expected))`

**Comment by joakim-hove** [comment]
OK - thank @pgdr - I will address these comments later today/tomorrow.

**Pull request merged**